desktop/apps/auction2/actions/filter.js

```diff
@@ -0,0 +1,397 @@
+export const TOGGLE_LIST_VIEW = 'TOGGLE_LIST_VIEW'
+export const UPDATE_AGGREGATED_ARTISTS = 'UPDATE_AGGREGATED_ARTISTS'
+export const UPDATE_AGGREGATED_MEDIUMS = 'UPDATE_AGGREGATED_MEDIUMS'
+export const UPDATE_ALL_FETCHED = 'UPDATE_ALL_FETCHED'
+export const UPDATE_ARTIST_ID = 'UPDATE_ARTIST_ID'
+export const UPDATE_ESTIMATE_DISPLAY = 'UPDATE_ESTIMATE_DISPLAY'
+export const UPDATE_ESTIMATE_RANGE = 'UPDATE_ESTIMATE_RANGE'
+export const UPDATE_INITIAL_MEDIUM_MAP = 'UPDATE_INITIAL_MEDIUM_MAP'
+export const UPDATE_IS_LAST_FOLLOWED_ARTISTS_PAGE = 'UPDATE_IS_LAST_FOLLOWED_ARTISTS_PAGE'
+export const UPDATE_MEDIUM_ID = 'UPDATE_MEDIUM_ID'
+export const UPDATE_NUM_ARTISTS_YOU_FOLLOW = 'UPDATE_NUM_ARTISTS_YOU_FOLLOW'
+export const UPDATE_PAGE = 'UPDATE_PAGE'
+export const UPDATE_SALE_ARTWORKS = 'UPDATE_SALE_ARTWORKS'
+export const UPDATE_SALE_ARTWORKS_BY_FOLLOWED_ARTISTS = 'UPDATE_SALE_ARTWORKS_BY_FOLLOWED_ARTISTS'
+export const UPDATE_SALE_ARTWORKS_BY_FOLLOWED_ARTISTS_TOTAL = 'UPDATE_SALE_ARTWORKS_BY_FOLLOWED_ARTISTS_TOTAL'
+export const UPDATE_SORT = 'UPDATE_SORT'
+export const UPDATE_TOTAL = 'UPDATE_TOTAL'
+
+// Action creators
+export function decrementFollowedArtistsPage () {
+  return {
+    type: DECREMENT_FOLLOWED_ARTISTS_PAGE
+  }
+}
+
+export function getArtworksFailure () {
+  return {
+    type: GET_ARTWORKS_FAILURE
+  }
+}
+
+export function getArtworksRequest () {
+  return {
+    type: GET_ARTWORKS_REQUEST
+  }
+}
+
+export function getArtworksSuccess () {
+  return {
+    type: GET_ARTWORKS_SUCCESS
+  }
+}
+
+export function fetchArtworks () {
+  return async (dispatch, getState) => {
+    const {
+      auctionArtworks: {
+        filterParams,
+        user
+      }
+    } = getState()
+
+    try {
+      dispatch(getArtworksRequest())
+
+      const { filter_sale_artworks } = await metaphysics({
+        query: filterQuery,
+        variables: filterParams,
+        req: {
+          user
+        }
+      })
+
+      if (filter_sale_artworks.hits && filter_sale_artworks.hits.length) {
+        analyticsHooks.trigger(
+          'auction:artworks:loaded',
+          { data: filter_sale_artworks.hits.map((sale_artwork) => sale_artwork.artwork._id) }
+        )
+      }
+
+      const aggregations = filter_sale_artworks.aggregations
+      const artistAggregation = aggregations.filter((agg) => agg.slice === 'ARTIST')
+      const mediumAggregation = aggregations.filter((agg) => agg.slice === 'MEDIUM')
+
+      dispatch(updateAggregatedArtists(artistAggregation[0].counts))
+      dispatch(updateAggregatedMediums(mediumAggregation[0].counts))
+      dispatch(updateInitialMediumMap(mediumAggregation[0].counts))
+      dispatch(updateTotal(filter_sale_artworks.counts.total))
+      dispatch(updateNumArtistsYouFollow(filter_sale_artworks.counts.followed_artists))
+      dispatch(updateSaleArtworks(filter_sale_artworks.hits))
+      dispatch(updateAllFetched())
+      dispatch(getArtworksSuccess())
```

Comment:
Rather than `dispatch`ing so many individual actions, why not create an action whose purpose is to set each of these things? Then the individual reducers would listen for that action.